### PR TITLE
#4220 support banning lot numbers with no product

### PR DESCRIPTION
**Pull request by metas-rc**
https://github.com/metasfresh/metasfresh/issues/4220

### CODE REVIEW

File: de.metas.swat/de.metas.swat.base/src/main/java/de/metas/product/LotNumberLockRepository.java [hunk 1]
```diff
@@ -58,10 +58,11 @@ public LotNumberLock getByProductIdAndLot(final int productId, final String lotN
 				.createQueryBuilder(I_M_Product_LotNumber_Lock.class)
 				.addOnlyActiveRecordsFilter()
 				.addOnlyContextClient()
-				.addEqualsFilter(I_M_Product_LotNumber_Lock.COLUMN_M_Product_ID, productId)
+				.addInArrayFilter(I_M_Product_LotNumber_Lock.COLUMN_M_Product_ID, productId, null)
 				.addEqualsFilter(I_M_Product_LotNumber_Lock.COLUMNNAME_Lot, lotNo)
+				.orderBy(I_M_Product_LotNumber_Lock.COLUMNNAME_M_Product_ID)
```
Comment (metas-ts) [inline de.metas.swat/de.metas.swat.base/src/main/java/de/metas/product/LotNumberLockRepository.java]: ok: this orders with nulls last

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/40-sw01_swat/5496860_sys_gh4220_LotNoLock_Product_NotMandatory.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-- 2018-06-29T17:27:07.163
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_Column SET IsMandatory='N',Updated=TO_TIMESTAMP('2018-06-29 17:27:07','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=559563
```
Comment (metas-ts) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/40-sw01_swat/5496860_sys_gh4220_LotNoLock_Product_NotMandatory.sql]: AFAIU we also need the DDL to make the physical column not-mandatory

Comment (metas-rc) [reply de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/40-sw01_swat/5496860_sys_gh4220_LotNoLock_Product_NotMandatory.sql]: Done

File: de.metas.swat/de.metas.swat.base/src/main/java/de/metas/product/LotNumberLockRepository.java (hunk 1 above)
Comment (metas-rc) [reply de.metas.swat/de.metas.swat.base/src/main/java/de/metas/product/LotNumberLockRepository.java]: yes

**Pull request merged**